modules/gcp/region.go

```diff
@@ -0,0 +1,188 @@
+func GetRandomRegionE(t *testing.T, approvedRegions []string, forbiddenRegions []string) (string, error) {
+	regionFromEnvVar := os.Getenv(regionOverrideEnvVarName)
+	if regionFromEnvVar != "" {
+		logger.Logf(t, "Using GCP region %s from environment variable %s", regionFromEnvVar, regionOverrideEnvVarName)
+		return regionFromEnvVar, nil
+	}
+
+	regionsToPickFrom := approvedRegions
+
+	if len(regionsToPickFrom) == 0 {
+		allRegions, err := GetAllGcpRegionsE(t)
+		if err != nil {
+			return "", err
+		}
+		regionsToPickFrom = allRegions
+	}
+
+	regionsToPickFrom = collections.ListSubtract(regionsToPickFrom, forbiddenRegions)
+	region := random.RandomString(regionsToPickFrom)
+
+	logger.Logf(t, "Using region %s", region)
+	return region, nil
+}
+
+// GetRandomZone gets a randomly chosen GCP zone. If approvedRegions is not empty, this will be a zone from the approvedZones
+// list; otherwise, this method will fetch the latest list of zones from the GCP APIs and pick one of those. If
+// forbiddenZones is not empty, this method will make sure the returned region is not in the forbiddenZones list.
+func GetRandomZone(t *testing.T, approvedZones []string, forbiddenZones []string) string {
+	zone, err := GetRandomZoneE(t, approvedZones, forbiddenZones)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return zone
+}
+
+// GetRandomZoneE gets a randomly chosen GCP zone. If approvedRegions is not empty, this will be a zone from the approvedZones
+// list; otherwise, this method will fetch the latest list of zones from the GCP APIs and pick one of those. If
+// forbiddenZones is not empty, this method will make sure the returned region is not in the forbiddenZones list.
+func GetRandomZoneE(t *testing.T, approvedZones []string, forbiddenZones []string) (string, error) {
+	zoneFromEnvVar := os.Getenv(zoneOverrideEnvVarName)
+	if zoneFromEnvVar != "" {
+		logger.Logf(t, "Using GCP zone %s from environment variable %s", zoneFromEnvVar, zoneOverrideEnvVarName)
+		return zoneFromEnvVar, nil
+	}
+
+	zonesToPickFrom := approvedZones
+
+	if len(zonesToPickFrom) == 0 {
+		allZones, err := GetAllGcpZonesE(t)
+		if err != nil {
+			return "", err
+		}
+		zonesToPickFrom = allZones
+	}
+
+	zonesToPickFrom = collections.ListSubtract(zonesToPickFrom, forbiddenZones)
+	zone := random.RandomString(zonesToPickFrom)
+
+	logger.Logf(t, "Using zone %s", zone)
+	return zone, nil
+}
+
+// GetAllGcpRegions gets the list of GCP regions available in this account.
+func GetAllGcpRegions(t *testing.T) []string {
+	out, err := GetAllGcpRegionsE(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return out
+}
+
+// GetAllGcpRegionsE gets the list of GCP regions available in this account.
+func GetAllGcpRegionsE(t *testing.T) ([]string, error) {
+	logger.Log(t, "Looking up all GCP regions available in this account")
+
+	// TODO - NewComputeServiceE creates a context, should we get that somehow
+	// or use a new one here?
+	ctx := context.Background()
```

Comment:
This seems cleaner for now. I still haven't seen the `context` actually used for anything these GCP functions so far, so my vote is to keep it simple and consider alternatives if/when necessary.

Reply:
Okay so leave as is?

Reply:
Yeah, let's leave as is. I was just affirming your decision.

Reply:
cool